packages/amplify-graphql-auth-transformer/src/__tests__/owner-auth.test.ts

```diff
@@ -481,6 +481,95 @@ describe('owner based @auth', () => {
     expect(out.schema).toMatchSnapshot();
   });
 
+  it('should not add duplicate implicit field when already included', async () => {
+    const authConfig: AppSyncAuthConfiguration = {
+      defaultAuthentication: {
+        authenticationType: 'AMAZON_COGNITO_USER_POOLS',
+      },
+      additionalAuthenticationProviders: [],
+    };
+    const validSchema = `
+      type Todo @model @auth(rules: [{ allow: owner }]) {
+        owner: String
+        content: String
+      }
+    `;
+
+    const out = testTransform({
+      schema: validSchema,
+      authConfig,
+      transformers: [new ModelTransformer(), new AuthTransformer()],
+    });
+    expect(out).toBeDefined();
+
+    validateModelSchema(parse(out.schema));
+    parse(out.schema);
+    expect(out.schema).toMatchSnapshot();
+  });
+
+  it('should successfully transform simple valid schema with implicit fields', async () => {
+    const authConfig: AppSyncAuthConfiguration = {
+      defaultAuthentication: {
+        authenticationType: 'AMAZON_COGNITO_USER_POOLS',
+      },
+      additionalAuthenticationProviders: [],
+    };
+    const validSchema = `
+      type Todo @model @auth(rules: [{ allow: owner }]) {
+        id: ID! @primaryKey
+        content: String
+      }
+    `;
+
+    const out = testTransform({
+      schema: validSchema,
+      authConfig,
+      transformers: [new ModelTransformer(), new AuthTransformer(), new PrimaryKeyTransformer()],
+      dataSourceStrategies: constructDataSourceStrategies(validSchema, {
+        name: `MYSQLStrategy`,
+        dbType: MYSQL_DB_TYPE,
+        dbConnectionConfig: {
+          databaseNameSsmPath: '/databaseNameSsmPath',
+          hostnameSsmPath: '/hostnameSsmPath',
+          passwordSsmPath: '/passwordSsmPath',
+          portSsmPath: '/portSsmPath',
+          usernameSsmPath: '/usernameSsmPath',
+        },
+      }),
```

Comment:
nit: we have a [test utility](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-transformer-test-utils/src/datasource-utils/utils.ts#L45) for constructing mock sql datasource strategies

```suggestion
      dataSourceStrategies: constructDataSourceStrategies(validSchema, mockSqlDataSourceStrategy(),
```

Reply:
If you see other cases where we're still manually constructing strategies, can you point them out in a backlog item for us to fix later?